specutils/analysis/flux.py

```diff
@@ -116,3 +124,78 @@ def _compute_equivalent_width(spectrum, continuum=1, regions=None):
     ew =  dx - (line_flux / continuum)
 
     return ew.to(calc_spectrum.spectral_axis.unit)
+
+def is_continuum_below_threshold(spectrum, threshold=0.01):
+    """
+    Determine if the baseline of this spectrum is less than the threshold.
+    I.e., an estimate of whether or not the continuum has been subtracted.
+
+    If the threshold is an `~astropy.units.Quantity` then compare the median
+    of the flux to the threshold.
+
+    If the threshold is a float then the spectrum's uncertainty will be
+    used or an estimate of the uncertainty. If the uncertainty is present then the
+    threshold is compared to the median of the flux divided by the
+    uncertainty.  If the uncertainty is not present then the threshold
+    is compared to the median of the flux divided by the
+    `~astropy.stats.median_absolute_deviation`.
+
+    Parameters
+    ----------
+    spectrum : `~specutils.spectra.spectrum1d.Spectrum1D`
+        The spectrum object over which the width will be calculated.
+
+    threshold: float or `~astropy.units.Quantity`
+        The tolerance on the quantification to confirm the continuum is
+        near zero.
+
+    Returns
+    -------
+    is_continuum_below_threshold: bool
+        Return True if the continuum of the spectrum is below the threshold, False otherwise.
+
+    """
+
+    flux = spectrum.flux
+    uncertainty = spectrum.uncertainty if hasattr(spectrum, 'uncertainty') else None
+
+    # Apply the mask if it exists.
+    if hasattr(spectrum, 'mask') and spectrum.mask is not None:
+        flux = flux[~spectrum.mask]
+        uncertainty = uncertainty[~spectrum.mask] if uncertainty else uncertainty
+
+    # If the threshold has units then the assumption is that we want
+    # to compare the median of the flux regardless of the
+    # existence of the uncertainty.
+    if hasattr(threshold, 'unit'):
+        return np.median(flux) < threshold
+
+    # If threshold does not have a unit, ie it is not a quantity, then
+    # we are going to calculate based on the S/N if the uncertainty
+    # exists.
+    if uncertainty:
+        return np.median(flux / uncertainty.quantity) < threshold
+    else:
+        return np.median(flux) / median_absolute_deviation(flux) < threshold
+
+def warn_continuum_below_threshold(threshold=0.01, check=conf.do_continuum_function_check):
+    """
+    Decorator for methods that should warn if the baseline
+    of the spectrum does not appear to be below a threshold.
+
+    The ``check`` parameter is based on the
+    `astropy configuration system <http://docs.astropy.org/en/stable/config/#adding-new-configuration-items>`_.
+    Examples are on that page to show how to turn off this type of warning checking.
+    """
+    def actual_decorator(function):
+        @wraps(function)
+        def wrapper(*args, **kwargs):
+            if check:
```

Comment:
Hmm. I might be confused but I think this isn't quite right - I think you want to do 
```suggestion
            if check():
```
because a configuration item can be changed at runtime.

Reply:
Or alternatively,
```suggestion
            if check() if callable(check) else check:
```
which would let you pass in True/False in addition to a configuration item

Reply:
Based on our conversation, this changed to an alternative form (that is correct :) ).